src/runtime/os_linux.go

```diff
@@ -139,3 +146,16 @@ func hardwareRand() (n uint64, ok bool) {
 //
 //export getrandom
 func libc_getrandom(buf unsafe.Pointer, buflen uintptr, flags uint32) uint32
+
+// int fcntl(int fd, int cmd, int arg);
+//
+// export fcntl
+func libc_fcntl(fd int, cmd int, arg int) (ret int)
+
+func fcntl(fd int32, cmd int32, arg int32) (ret int32, errno int32) {
+	ret = int32(libc_fcntl(int(fd), int(cmd), int(arg)))
+	if ret < 0 {
+		return 0, int32(uintptr(*libc_errno_location()))
```

Comment:
Why this double cast?

```suggestion
		return 0, *libc_errno_location()
```